packages/devtools_app/lib/src/charts/flutter/flame_chart.dart

```diff
@@ -14,6 +18,162 @@ const double sectionSpacing = 15.0;
 const double topOffset = rowHeightWithPadding;
 const double sideInset = 70.0;
 
+abstract class FlameChart<T, V extends FlameChartNodeDataMixin>
```

Comment:
Generally, Flutter design is to create widgets to compose instead of extend.

Reply:
Noted. Added a todo to cleanup later. I have some pending work for the full timeline flame chart that this refactor would disturb, so I'd rather cleanup after the full timeline code lands

Reply:
SGTM.